packages/client/src/bitcoinEncode.ts

```diff
@@ -32,22 +34,44 @@ class Header {
 
 // FieldsV0 Class
 class FieldsV0 {
-  receiver: string;
-  payload: Uint8Array;
-  revertAddress: string;
-
-  constructor(receiver: string, payload: Uint8Array, revertAddress: string) {
-    if (!isAddress(receiver)) {
-      throw new Error("Invalid receiver address");
-    }
+  receiver: Address;
+  payload: Buffer;
+  revertAddress: BtcAddress;
+
+  constructor(receiver: Address, payload: Buffer, revertAddress: BtcAddress) {
     this.receiver = receiver;
     this.payload = payload;
     this.revertAddress = revertAddress;
   }
 }
 
+/**
+ * Encodes data for a Bitcoin transaction
+ * @param receiver - The address of the receiver
+ * @param payload - The payload to be sent
+ * @param revertAddress - Bitcoin address to revert funds to in case of failure
+ * @param opCode - The operation code (defaults to DepositAndCall)
+ * @param encodingFormat - The encoding format (defaults to ABI)
+ * @returns The encoded data as a hex string
+ */
+export const bitcoinEncode = (
```

Comment:
I'm wondering if this is the best place to have this type of function, specially considering this is not being exported from the client, wouldn't it be more fitting on the root `utils` folder?

Reply:
Moved it to utils: bf2cc3c

In general, I think we should consider separating functions intended for public use from more internal functions: https://github.com/zeta-chain/toolkit/issues/322